pytensor/graph/basic.py

```diff
@@ -970,6 +970,83 @@ def applys_between(
     )
 
 
+def condition_subset(
```

Comment:
For the above graph (I can't draw) the 3d input will be also returned, because it does not depend on the condition and is a part of output graph. Yes, this node could be computed from the parent node circled in red, but it will never be a requirement to recompute since only the condition nodes are supposed to change. What about the corner case with no conditions, it may make sense, I agree.

Reply:
The purpose of the condition subset is to return an input->output subgraph that has to be reconstructed if all or any of the inputs change. I later realized that I can safely drop constants that are not marked as inputs in the return

Reply:
```
Examples
    --------
    The returned nodes marked in (parenthesis), condition nodes are ``c``, output nodes are ``o``

    * No conditions

    .. code-block::
    
        n - n - (o)

    * One condition
    
    .. code-block::
    
        n - (c) - o

    * Two conditions where on depends on another
    
    .. code-block::
    
        (c) - c - o

    * Additional inputs are present
    
    .. code-block::
    
        (c) - c - o
         (n) -'

    * Disconnected condition
    
    .. code-block::
    
        (c) - c - o
         c 

    * Disconnected output is present
    
    .. code-block::
    
        (c) - c - o
        (o)
```

Reply:
For the replace functionality I would also want to know if a condition was dropped because it is inside the computational graph and depends on another conditions or because it was disconnected from the output. Maybe the return could look like 

```
boundary, inside, disconnected
```

It would also be sufficient to merge boundary+inside but I have to think a little bit more

Reply:
the inside condition being included does not seem to create any fundamental problems, however. If any node depends on it it would also depend on some another condition